internal/contracts/manager.go

```diff
@@ -572,6 +573,22 @@ func (cm *contractManager) GetContractListenerByNameOrID(ctx context.Context, na
 	return listener, nil
 }
 
+func (cm *contractManager) GetContractListenerByNameOrIDWithStatus(ctx context.Context, nameOrID string) (enrichedListener *core.ContractListenerWithStatus, err error) {
+	listener, err := cm.GetContractListenerByNameOrID(ctx, nameOrID)
+	if err != nil {
+		return nil, err
+	}
+	status, err := cm.blockchain.GetContractListenerStatus(ctx, listener.BackendID)
+	if err != nil {
+		return nil, err
```

Comment:
Just an observation that if you use `fetchstatus` this means that if the connector is not available or returns a `404`, you will get an error. I wonder @shorsher if that's the behavior we want, or we should pop a `status.error` field in there instead.

Reply:
Yep, agreed. Thanks for the suggestion 🙏🏻

Reply:
I think I made this way too complicated, but really struggled with the right way to get `status.error` to show up correctly while also including the `checkpoint` as a `JSONAny` type 😅 


Before, `cm.blockchain.GetContractListenerStatus(ctx, listener.BackendID)` would return a `JSONAny` of:
```json
{
  "checkpoint": {
     "block": 0,
     "transactonIndex": 0,
     "logIndex": 0
   },
  "catchup": false
}
```

This was great, because I could make the `ContractListenerWithStatus` struct look like:
```golang
type ContractListenerWithStatus struct {
	ContractListener
	Status *fftypes.JSONAny `ffstruct:"ContractListenerWithStatus" json:"status,omitempty" ffexcludeinput:"true"`
}
```

and set what is returned from `GetContractListenerStatus()` to `Status`, and everything would be 👍🏻. I was not sure how to update `ContractListenerWithStatus` to also add an `error` key within `status`.

Reply:
thanks for the help here, should have thought of using `interface{}`. Code is way better than what I originally came up with.